internal/services/subscription/subscription_resource.go

```diff
@@ -257,11 +257,11 @@ func resourceSubscriptionUpdate(d *pluginsdk.ResourceData, meta interface{}) err
 	locks.ByName(id.AliasName, SubscriptionResourceName)
 	defer locks.UnlockByName(id.AliasName, SubscriptionResourceName)
 	resp, err := aliasClient.AliasGet(ctx, *id)
-	if err != nil || resp.Model == nil || resp.Model.Properties == nil {
+	if err != nil || resp.Model == nil || resp.Model.Properties == nil || resp.Model.Properties.SubscriptionId == nil {
 		return fmt.Errorf("could not read Subscription Alias for update: %+v", err)
 	}
 
-	subscriptionId, err := commonids.ParseSubscriptionID(*resp.Model.Properties.SubscriptionId)
+	subscriptionId, err := commonids.ParseSubscriptionID(fmt.Sprintf("/subscriptions/%s", *resp.Model.Properties.SubscriptionId))
```

Comment:
we should never be hard-coding Resource IDs like this, instead we can use:

```suggestion
	subscriptionId := commonids.NewSubscriptionID(*resp.Model.Properties.SubscriptionId)
```